src/execute.rs

```diff
@@ -422,7 +432,38 @@ pub fn run(cli: Cli) -> Result<(), Box<dyn Error>> {
             ProofSystem::KZG => {
                 let proof = Snark::load::<KZGCommitmentScheme<Bn256>>(&proof_path, None, None)?;
                 let code = DeploymentCode::load(&deployment_code_path)?;
-                evm_verify(code, proof)?;
+                evm_verify(code, proof.clone())?;
+
+                if sol_code_path.is_some() {
+                    let (anvil, client) = setup_eth_backend().await;
```

Comment:
if you  want to -- you can bundle up this section as a standalone function -- whatever your preference is. It might be useful to do so later on; especially if `pyezkl` calls components of this

Reply:
:+1: done, check out https://github.com/zkonduit/ezkl/commit/d6abd3f6b5614364437a8cd83f25a1c903f8edfd#diff-b9cde71f35f4266fc72cee1dccc59f632255fa628e923d8cbd9c6429b7a1dcf6R46